src/basic/audit-util.c

```diff
@@ -68,6 +70,67 @@ int audit_loginuid_from_pid(pid_t pid, uid_t *uid) {
         return 0;
 }
 
+static int try_audit_request(int fd) {
+        struct iovec iov;
+        struct msghdr mh;
+        ssize_t n;
+
+        assert(fd >= 0);
+
+        struct {
+                struct nlmsghdr hdr;
+                struct nlmsgerr err;
+        } _packed_ msg = {
+                .hdr.nlmsg_len = NLMSG_LENGTH(0),
+                .hdr.nlmsg_type = AUDIT_GET_FEATURE,
+                .hdr.nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK,
+        };
+        iov = (struct iovec) {
+                .iov_base = &msg,
+                .iov_len = msg.hdr.nlmsg_len,
+        };
+        mh = (struct msghdr) {
+                .msg_iov = &iov,
+                .msg_iovlen = 1,
+        };
+
+        if (sendmsg(fd, &mh, MSG_NOSIGNAL) < 0) {
+                log_debug_errno(errno, "Failed to send AUDIT_GET_FEATURE request, ignoring: %m");
+                return 0;
```

Comment:
so the error handling is a bit confusing. i.e. it's fine eating up errors, but please decide whether the caller or the callee eats them up. Here you eat them up in both places, mostly?

I think I'd let this function propagate them to the caller, and then let the caller eat them up